docs/src/rules/preserve-caught-error.md

```diff
@@ -0,0 +1,62 @@
+---
+title: preserve-caught-error
+rule_type: suggestion
+further_reading:
+- https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause
+- https://nodejs.org/api/errors.html#errorcause
+- https://github.com/tc39/proposal-error-cause/blob/main/README.md
+- https://dev.to/amnish04/never-lose-valuable-error-context-in-javascript-3aco
+---
+
+JavaScript developers often re-throw errors in `catch` blocks to add context but forget to preserve the original error, resulting in lost debugging information.
+
+Using the `cause` option when throwing new errors helps retain the **original error** and maintain complete **error chains**, which improves debuggability and traceability.
+
+## Rule Details
+
+This rule enforces the use of the [`cause`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Error/cause) property when throwing a new error inside a `catch` block.
+
+Checks for all built-in [error types](https://github.com/microsoft/TypeScript/blob/main/src/lib/es2022.error.d.ts) that support passing a `cause`.
+
+Examples of **incorrect** code for this rule:
+
+```js
+/* eslint preserve-caught-error: "error" */
+
+// Not using the `cause` option
+try {
+    // ...
+} catch (error) {
+    throw new Error("Something went wrong: " + error.message);
+}
+
+// Ignoring the caught error at the parameter level
+try {
+	doSomething();
+} catch {
+	throw new TypeError("Something went wrong");
```

Comment:
Upon thinking about this further, I think this case should be valid as it seems like the intent is to throw away the error. After all, if the rule is `preserve-caught-error`, the error here isn't actually caught.

Maybe we could add an option to allow catching this pattern as well if people want?

@mdjermanovic @fasttime what do you think?

Reply:
> I think it's a good idea to have an option to report throw statements in catch blocks that don't bind the error, and it's fine if that defaults to `false`.

Sounds good to me.